### PR TITLE
#12414 - Generate enum values based on country locale in Developer view

### LINKED ISSUE #12414: Dev mode generates data with invalid enum values for the server country
### Bug Description

There are currently three scenarios that I observed regarding the failure of sending cases to SurvNet on the case page or in bulk edit mode.

### Reproduction Steps

I: Sending cases to SurvNet fails from bulk edit mode but is successful from the case page:
1. From the 'Configurations' directory, 'Developer' tab, generate 100 Covid-19 cases;
2. Navigate to the 'Cases' directory and enter bulk edit mode;
3. Select some of the cases generated (either under or over 40);
4. From 'Bulk actions' select 'Send to reporting tool';
Actual behavior: The bulk action fails and an error is displayed.
5. After the error appeared, open the first case in the list;
6. Click on 'Send to reporting tool'.
Actual behavior: the case is successfully sent.

II: Sending cases to SurvNet fails from bulk edit mode and from the case page: 
1. From the 'Configurations' directory, 'Developer' tab, generate 100 Covid-19 cases;
2. Navigate to the 'Cases' directory and enter bulk edit mode;
3. Select some of the cases generated (either under or over 40);
4. From 'Bulk actions' select 'Send to reporting tool';
Actual behavior: The bulk action fails and an error is displayed.
5. After the error appeared, open the first case in the list;
6. Click on 'Send to reporting tool'.
Actual behavior: Sending the case to reporting tool fails.

Server logs when sending cases to SurvNet fails from bulk edit mode:
[Under 40 entries selected error in bulk edit mode.txt](https://github.com/SORMAS-Foundation/SORMAS-Project/files/12312498/Under.40.entries.selected.error.in.bulk.edit.mode.txt)

[Over 40 entries selected error in bulk edit mode.txt](https://github.com/SORMAS-Foundation/SORMAS-Project/files/12312522/Over.40.entries.selected.error.in.bulk.edit.mode.txt)

Server logs when sending cases to SurvNet fails from the case page:
[Error when sending only 1 case to SurvNet.txt](https://github.com/SORMAS-Foundation/SORMAS-Project/files/12312545/Error.when.sending.only.1.case.to.SurvNet.txt)

### Expected Behavior

All cases that are not marked as 'Do not share with reporting tool' should be able to be successfully sent to SurvNet from the case page and bulk edit mode, regardless of whether under or over 40 entries are selected.

### Affected Area(s)

Web app

### SORMAS Version

1.87.0, 1.88.0

### Android version/Browser

Chrome

### User Role(s)

Admin+NatUser

### Server URL

test-de3, test-de2, test1402

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by carina29**
Fixes #12414

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12536

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12536

File: sormas-api/src/main/java/de/symeda/sormas/api/contact/QuarantineType.java [hunk 1]
```diff
@@ -27,8 +27,7 @@ public static boolean isQuarantineInEffect(QuarantineType quarantineType) {
 		return QUARANTINE_IN_EFFECT.contains(quarantineType);
 	}
 
-	@Override
-	public String toString() {
+	public String toEnumCaption() {
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/contact/QuarantineType.java:30]: Please revert this change, the method is required like this

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12536

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12536

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**